octoprint_mrbeam/migrate.py

```diff
@@ -627,6 +652,56 @@ def disable_gcode_auto_deletion(self):
         if not self.plugin.isFirstRun():
             self.plugin._settings.set_boolean(["gcodeAutoDeletion"], False)
 
+    def fix_s_series_mount_manager(self):
+        """
+        fixes a problem with the images before 19.7.2021
+        the rc.local file was missing the clear command for the mount_manager
+        this replaces the rc.local file with the one containing this row
+        """
+        self._logger.info("start fix_s_series_mount_manager")
+        src_rc_local = os.path.join(
+            __package_path__, self.MIGRATE_FILES_FOLDER, "fix_s_series_mount_manager"
+        )
+        dst_rc_local = "/etc/rc.local"
+        if exec_cmd("sudo cp {src} {dst}".format(src=src_rc_local, dst=dst_rc_local)):
+            self._logger.info("rc.local file copied to %s", dst_rc_local)
+
+        systemdfiles = (
+            "mount_manager_remove.service",
+            "mount_manager_remove_before_octo.service",
+            "mount_manager_add.service",
+        )
+        dst = "/etc/systemd/system"
+        self._logger.info("copy files")
+        for systemdfile in systemdfiles:
+            success = True
+            src = os.path.join(__package_path__, self.MIGRATE_FILES_FOLDER, systemdfile)
+            if not exec_cmd("sudo cp {src} {dst}".format(src=src, dst=dst)):
+                success = False
+            self._logger.info("enable %s", systemdfile)
+            if not exec_cmd("sudo systemctl daemon-reload"):
+                success = False
+            if not exec_cmd(
+                "sudo systemctl enable {systemdfile}".format(systemdfile=systemdfile)
+            ):
+                success = False
+            if success:
+                self._logger.info("successfully created ", systemdfile)
```

Comment:
This is a little python hack:
```python
success &= exec_cmd(...)
```
:)
But you do not seem to do anything if it is not successful?

Reply:
Ahh I was looking for something like that :)

Reply:
```python
if exec_cmd("sudo cp {src} {dst}".format(src=src, dst=dst)):
   and exec_cmd("sudo systemctl daemon-reload"):
   and exec_cmd("sudo systemctl enable {systemdfile}".format(systemdfile=systemdfile)
):
    self._logger.info("successfully created ", systemdfile)
```

Reply:
```python
systemdfiles = (
    (True, "mount_manager_remove.service"),
    (True, "mount_manager_remove_before_octo.service"),
    (False, "mount_manager_add.service"),
)
for enable, systemdfile in systemdfiles:
  ...

Reply:
```python
if exec_cmd("sudo cp {src} {dst}".format(src=src, dst=dst))
   and exec_cmd("sudo systemctl daemon-reload")
   and (not enable or exec_cmd("sudo systemctl enable {}".format(systemdfile))
):
    self._logger.info("successfully created ", systemdfile)
```